app/controllers/v0/user/mhv_user_accounts_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module V0
+  module User
+    class MHVUserAccountsController < ApplicationController
+      service_tag 'identity'
+
+      before_action :set_mhv_user_account, only: :show
+      rescue_from MHV::UserAccount::Errors::UserAccountError, with: ->(e) { render_errors(e.message) }
+
+      def show
+        return render_errors('not_found', status: :not_found) if @mhv_user_account.blank?
+
+        log_action('success')
+        render json: MHVUserAccountSerializer.new(@mhv_user_account).serializable_hash, status: :ok
+      end
+
+      private
+
+      def set_mhv_user_account
+        @mhv_user_account = current_user.mhv_user_account
+      end
+
+      def render_errors(error_message, status: :unprocessable_entity)
+        log_action('error', error_message)
+
+        errors = error_message.split(',').map { |m| { detail: m.strip } }
+        render json: { errors: }, status:
+      end
+
+      def log_action(result, error_message = nil)
+        payload = { error_message:, user_account: @mhv_user_account, icn: current_user.icn }.compact
+        Rails.logger.info("[User][MHVUserAccountsController] #{action_name} #{result}", payload)
```

Comment:
I'm wondering if we need the `payload` here (except the error message), we'll be able to get the rest of the info from the rest of the log in the request, and I'm not sure if we should log the whole `mhv_user_account` object, seems like it could log something unintentional if more is added to it

Reply:
yeah good call. removed